boa/src/builtins/json/mod.rs

```diff
@@ -149,15 +149,16 @@ pub fn stringify(_: &mut Value, args: &[Value], ctx: &mut Interpreter) -> Result
         for field in fields {
             if let Some(value) = object
                 .get_property(&ctx.to_string(&field)?)
-                .map(|prop| prop.value.as_ref().map(|v| v.to_json()))
+                .map(|prop| prop.value.as_ref().map(|v| v.to_json(ctx)))
                 .flatten()
```

Comment:
Here we can use the `Option::and_then` method so we avoid the `.flatten()`
```suggestion
                .and_then(|prop| prop.value.as_ref().map(|v| v.to_json(ctx)))
```